src/System.Buffers.Experimental/System/Range.cs

```diff
@@ -88,5 +89,77 @@ IEnumerator<int> IEnumerable<int>.GetEnumerator()
 
         IEnumerator IEnumerable.GetEnumerator()
             => GetEnumerator();
+
+        public bool Contains(int value)
+        {
+            if (!IsBound) throw new InvalidOperationException("Unbound ranges cannot contain");
+            return value >= First && value < End;
+        }
+
+        /// <summary>
+        /// Returns true if this Range is a valid range for a zero based index of sspecified length.
+        /// </summary>
+        /// <param name="length">zero based length.</param>
+        /// <returns></returns>
+        public bool IsValid(int length)
+        {
+            if (First == UnboundedFirst)
+            {
+                if (End == UnboundedEnd) return true;
+                return End <= length;
+            }
+            else // First is bounded
+            {
+                if (First < 0) return false;
+                if (End == UnboundedEnd) return First <= length;
+                if (First > length) return false;
+                return End <= length;
+            }
+
+            throw new NotImplementedException();
+        }
+
+        public Range Bind(int length)
```

Comment:
Can you please add a comment here explaining what Bind is meant to do?

Reply:
will do